### ISSUE TITLE
Hyperparameters for PPO and request for suggestions

### ISSUE THREAD

**Issue by artemi8** [Procuste34/Landing-Starships#1]
I see you have mentioned you used out of the box PPO algorithm, just wanted to confirm if these are the [hyperparameters](https://github.com/Unity-Technologies/ml-agents/blob/main/docs/Training-ML-Agents.md#behavior-configurations) that was used when you trained it. Also what other algorithm do you suggest for environment like this, I am trying to land [Pyflyt Rocker Lander](https://taijunjet.com/PyFlyt/documentation/gym_envs/rocket_landing_env.html) using RL and mine is still a WIP. Any suggestions would be really helpful! Thanks

Great work by the way!

**Comment by alxndrTL** [comment]
Thank you for the comment!
The exact hyperparameters can be found [here](https://github.com/alxndrTL/Landing-Starships/tree/master/config) (basically, it is a three-layers neural network (2 hidden layers of 128 units), a learning rate of 3.0E-4 and a batch size of 64 (and then increased later at 128))

A few things to note :
- the reward function is very simple, +1 for a correct landing, 0 else. if you look on the internet, people come up with fancy reward functions but it is subject to reward hacking and so I just used this simple one
- ofc, as the reward function is very sparse, I had to use curriculum learning : start with the rocket starting from not too high, standing still, etc.. Then, as training progresses, the rocket is spawned higher, with angular speed, etc...
- I also employed behavioral cloning **at the beginning only** to further guide the RL agents towards getting the rewards. For the first stage of the curriculum learning (easiest stage), I observed that it was still hard for random RL agents to actually get a reward. So I recorded myself (using ML Agents) landing rockets on this first learning stage. Certainly, with a bit more work, this behavioral cloning could have been avoided.

I guess PPO is still a good choice for these types of tasks
Hope this helps!
